### ISSUE TITLE
onResize elements disappear in Safari

### ISSUE THREAD

**Issue by WPThemer** [julianlloyd/scrollReveal.js#126]
Hi, I really love your plugin, I was using animate.css with wow.js but encountered a problem with parallax (stellar) with FIXED background in Safari, so I decided to go with your plugin. It works perfectly, but I am trying to figure out how to initialize the plugin again if someone resizes the browser.

In Firefox it works perfectly if browser is resized, but in Safari (ver 5.7) if you have not scrolled down and in the meantime browser is resized to small screen and scrolled down or even again resized to full browser, the elements containing the reveal properties disappear...

I am using it as:
```JavaScript
jQuery( document ).ready( function ( $ ) {
     window.sr = new scrollReveal();	
});
```
I also have a resize / onload as well:
```JavaScript
jQuery( window ).on( "load resize", function ( e ) {
...
});
```
How can re-initialize the plugin when the screen is resized ?.. any input would be appreciated as I have to finish this asap... 

Thanks once again.

Regards,

@WPThemer

**Comment by WPThemer** [comment]
Pheww....after countless hours of debugging, fixing elements, dropping scripts etc etc... finally I realized that this Safari the CRAP has a repaint bug.. so basically it ignore elements and doesnt not repaint them, which causes the elements hide..

while using stellar and at the same time elements have z-index and negative margin, you are going to face this problem, below is the fix for safari repaint (hidden) elements issue..

document.getElementById('element').style.display='none';
document.getElementById('element').offsetHeight;
document.getElementById('element').style.display='';

atleast works in my very complex and heavy HTML page... 
now time to play actually with this amazing plugin to discover its potential :) 

thanks for this plugin which has saved me headaches from animate.css for Safari..

**Comment by julianlloyd** [comment]
Hey @WPThemer, I’m glad you’ve worked out your issue and I appreciate the kind words.

I’m working on a "Working With 3rd Parties" section in the Wiki, I’m not sure I fully understand the issue you ran into, but perhaps we could simply describe the problem and solution for posterity.

**Comment by WPThemer** [comment]
Hi Julian, I will try to explain a bit the problem and the fixed I discovered...

I am using stellar.js and scrollReveal.js together, but I had a long page for demo which contained a lot of HTML + Javascript, so basically a lot of code.

The Error:
It worked fine in IE9+ and Firefox, but in Safari 5.7.1, it didn't. The website is responsive, so whenever I am adding any code, I resize the browse to see how it looks like, but in Safari, when I was resizing the browser window, it was kind of hiding the elements (disappearing)  with scrollReveal.js attribute, hence messing up the whole layout.

The Bug:
There is a bug in Safari 5.7.1 that if there are a lot of script running at the same time, it does not repaint the elements on resize and ignore them. So I had to manually repaint the area to show those hidden elements, I found the above solution from Stackoverflow to fight with the repaint issue. The code below should be applied to the effected elements on jQuery window resize..

document.getElementById( 'element' ).style.display = 'none';
document.getElementById( 'element' ).offsetHeight;
document.getElementById( 'element' ).style.display = '';

so now whenever I am resizing the window, it works perfectly.

ONE more ISSUE / BUG thing that I would like to mention here for those are using these 2 amazing plugins together, most of the programmers are having problem with using animate.css (with combination of wow.js) and stellar, if they set the background-attachment:fixed; it breaks badly in Safari 5.1.7 I did not find any solution to that after spending countless hours, so finally I moved to scrollReveal.js

After I fixed the Safari re-paint issue, another issue arrived, again all the background images (parallaxed by Stellar) were not loading after the scrollReveal,js was used only if RESET keyword was used, I looked at he scrollReveal.js code and found out the same issue people were having with animate.css and he was the following:

-webkit-perspective: 1000;
and
-webkit-backface-visibility: hidden;

so once I removed them from transition='' and reset='' variables, it works perfectly. I hope it saves someones times working with complex websites..

.. and THANK YOU for this plugin, it has saved me a lot of SIZE and headache :) cheers
